bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigTomlProvider.java

```diff
@@ -221,64 +226,94 @@ private Object getPrimitiveTomlValue(Module module, VariableKey key) {
     private int validateAndGetByteValue(Object tomlValue, String variableName) {
         int byteValue = ((Long) tomlValue).intValue();
         if (!isByteLiteral(byteValue)) {
-            throw new ConfigException(String.format(INVALID_BYTE_RANGE, variableName, tomlValue));
+            throw new ConfigTomlException(String.format(INVALID_BYTE_RANGE, variableName, tomlValue));
         }
         return byteValue;
     }
 
     private TomlNode getTomlNode(TomlNode tomlValue, String variableName, Type type) {
         TomlType tomlType = tomlValue.kind();
         if (tomlType != TomlType.KEY_VALUE) {
-            throw new ConfigException(String.format(INVALID_TOML_TYPE, variableName, type,
-                                                    getTomlTypeString(tomlValue)));
+            throw new ConfigTomlException(String.format(INVALID_TOML_TYPE, variableName, type,
+                    getTomlTypeString(tomlValue)), tomlValue);
         }
         tomlValue = ((TomlKeyValueNode) tomlValue).value();
         tomlType = tomlValue.kind();
         if (tomlType != getEffectiveTomlType(type, variableName)) {
-            throw new ConfigException(String.format(INVALID_TOML_TYPE, variableName, type,
-                                                    getTomlTypeString(tomlValue)));
+            throw new ConfigTomlException(String.format(INVALID_TOML_TYPE, variableName, type,
+                    getTomlTypeString(tomlValue)), tomlValue);
         }
         return tomlValue;
     }
 
-    private static TomlTableNode getConfigurationData(Path configFilePath) {
+    private static TomlTableNode getConfigurationData(Path configFilePath, boolean hasRequired) {
         if (!Files.exists(configFilePath)) {
-            return null;
+            if (hasRequired) {
+                throw new ConfigTomlException(String.format(CONFIG_FILE_NOT_FOUND, configFilePath));
```

Comment:
If user provides only cli configs, then not having toml is not a error. any reason why wee need this check?

Reply:
These checks are depending on required variables. If we are to remove them after supporting cli args, then these toml related errors will become less informative. 
Eg: 
file not exists error - 
if the modules contain any required variables and the config file does not exist. 

After CLI support, we can remove these checks and provide only the value required error at the end. 
Can we do that?

Reply:
+1 to change it after we fix the cli side.